pkg/reconciler/managed/reconciler.go

```diff
@@ -347,7 +354,7 @@ type ExternalClientFns struct {
 	ObserveFn    func(ctx context.Context, mg resource.Managed) (ExternalObservation, error)
 	CreateFn     func(ctx context.Context, mg resource.Managed) (ExternalCreation, error)
 	UpdateFn     func(ctx context.Context, mg resource.Managed) (ExternalUpdate, error)
-	DeleteFn     func(ctx context.Context, mg resource.Managed) error
+	DeleteFn     func(ctx context.Context, mg resource.Managed) (ExternalDelete, error)
```

Comment:
Changing this return signature will be a breaking change for all provider implementations.

Reply:
Yes, when providers update to the v1.17 version of crossplane-runtime, they will need to update their `ExternalClient.Delete` implementations to now return an `ExternalDelete` struct, which will very likely be empty if they have no additional details to provide about their delete operation.  Example from provider-kubernetes: https://github.com/jbw976/provider-kubernetes/commit/4a9146acc2b2444517ccc66f43cb3a2aa650a2e1#diff-475eea76845069ccefe80a5e3e03cf95a235099281bb9bcc874d19f243ed8e2c 

As discussed, you were on the fence and I'm erring on the side of consistency for all 3 CUD operations.